### ISSUE TITLE
Support embedding Text test cases into Markdown files

### ISSUE THREAD

**Issue by vairamsvsjdo** [robotframework/robotframework#2610]
Many source control systems including bitbucket support markdown natively. Adding markdown syntax will be much helpful.

**Comment by pekkaklarck** [comment]
Could you clarify where and how you'd like to use Markdown?

**Comment by vairamsvsjdo** [comment]
Markdown is supported in Visual Studio Team Services, Team Foundation Server and Bitbucket. There is no support for reStructuredText.
This will be more helpful to add "Acceptance Documentations" available naturally. Without which it adds additional dependencies to make documentations and scripts available as HTML, PDF documents.

**Comment by Tset-Noitamotua** [comment]
@vairamsvsjdo Are talking about saving robot test as MarkDown (`.md`) files instead of `.robot` or `.txt`?

**Comment by Tset-Noitamotua** [comment]
As we don't have unlimited history on our RF Slack channel I've saved the [Slack ](https://robotframework.slack.com/files/tset-noitamotua/F3VF9BCG4/mdreader_py____version_0_0_1_.py)into a [Gist][1].

[1]: https://gist.github.com/Tset-Noitamotua/75d15a2beb9ab6f1931d3871172ebbbf

**Comment by vairamsvsjdo** [comment]
@Tset-Noitamotua that is exactly I was looking for. @pekkaklarck  please add this into robotframework support. It works excellent. I have about 50 test script files embedded with acceptance criteria documentations and all work perfect.

**Comment by pekkaklarck** [comment]
OK, you meant creating test cases in Markdown files similarly as you can create tests in reStructuredText files nowadays. I'm totally fine with that idea and, as @Tset-Noitamotua commented, there's already been discussion about that earlier on different forums. You could have also meant using Markdown when defining library documentation which is a valid proposal too.

What needs to be done to actually get the requested Markdown support into the framework is proving a pull request. The Gist by @Tset-Noitamotua is a good start but needs a little cleanup. A bigger task is creating tests and wring documentation. If there's anyone interested, please first take a look at the CONTRIBUTING.rst file in the project root. I'm willing to help here or on Slack too.

**Comment by vairamsvsjdo** [comment]
@pekkaklarck will do that appropriately.
Doc for libraries 
Doc,
 code for PEP8 standard
Support for HTML test cases 
Support for text , tsv test cases
Are features in that support comes to my mind.
Identify others please.

**Comment by pekkaklarck** [comment]
Better to have just one feature per issue. This issue covers adding support for test cases in MD files, supporting MD in library docs requires another issue. That is likely to be a bit more complicated feature to implement too.

**Comment by pekkaklarck** [comment]
PR #2621 looks pretty good. I need to look could it be included into 3.0.3.

**Comment by pekkaklarck** [comment]
RF 3.0.3 has been delayed a lot. Now we want it out ASAP and decided to only include bug fixes in it. I'll thus descope this enhancement to RF 3.1 which we plant to start developing right after RF 3.0.3 is out.

**Comment by pekkaklarck** [comment]
@vairamsvsjdo, do you still have interest to work with this issue? Getting this into RF 3.1 would be great.

**Comment by vairamsvsjdo** [comment]
@pekkaklarck 
Yes, will work on this next week. What is your timeline? I have few other tasks to take care in OSS next week.

**Comment by pekkaklarck** [comment]
Target for 3.1 is early June.

**Comment by pekkaklarck** [comment]
No interest lately. Removing from the long overdue RF 3.1 scope.

**Comment by Tset-Noitamotua** [comment]
What is the status of this one? I would like to pick it up again.

What needs to be done to get it into official release?

**Comment by pekkaklarck** [comment]
This issue is a bit confusing because there's discussion about different Markdown related enhancements and the initial description is not too clear about what's proposed. I thus submitted a new issue with a clearer description and close this as its duplicate. Good news is that we are adding Markdown support to Robot in general in RF 7.5 and there's a possibility that we add support for embedding tests/tasks in Markdown files as well.
